nix/build.nix

```diff
@@ -1,27 +1,40 @@
-{ stdenvNoCC, gitignoreSource, texlive }:
+{ nixpkgs ? (import ../nix/sources.nix).nixpkgs
+, pkgs    ?  import nixpkgs { }
```

Comment:
In a file describing a single derivation it's considered poor style to just take things from `pkgs`. This breaks overridability of inputs. There are [some resoures](https://nix-tutorial.gitlabpages.inria.fr/nix-tutorial/experiment-packaging.html) about how to use `callPackage` effectively. The changes you introduced are somewhat misguided. :)

When you've split the PRs, we can jump on a call to fix the Nix expressions for `latexmk`.